### DISCUSSION TITLE
Cannot get product prices in bm extensions

### DISCUSSION THREAD

**Discussion by TonyRlq** [SalesforceCommerceCloud/sfcc-ci#526] [Team Posts]
I am trying to get product prices in bm extensions,  but they are returning as 0, I cant get Price Model  and Price Book, Does anybody know why ?

**Comment by tparikh2** [comment]
Are you trying fetch product price for master or variant?

Thanks & Regards,
Tushar Parikh.

+91-909-901-3316

**Comment by TonyRlq** [comment]
> Are you trying fetch product price for master or variant?
> 
> Thanks & Regards,
> Tushar Parikh.
> 
> +91-909-901-3316
> …

Both master and variant, and I can get the prices of these products at storefront.

Best Regards,
Tony

**Comment by zlatinz** [comment]
In BM context, you do not have price book assigned to the "site". 
You need to use dw.catalog.PriceBookMgr to get the list of pricebooks e.g. using getAllPriceBooks()
Then from this list, you need to get dw.catalog.ProductPriceModel from the dw.catalog.Product object and use this ProductPriceModel to get the prices from the pricebooks. 
You may want to do some kind of configuration that actually determines which are the pricebooks you want to use.

**Comment by TonyRlq** [comment]
> In BM context, you do not have price book assigned to the "site".
> You need to use dw.catalog.PriceBookMgr to get the list of pricebooks e.g. using getAllPriceBooks()
> Then from this list, you need to get dw.catalog.ProductPriceModel from the dw.catalog.Product object and use this ProductPriceModel to get the prices from the pricebooks.
> You may want to do some kind of configuration that actually determines which are the pricebooks you want to use.

I got the list of pricebooks by using PriceBookMgr, then through PriceBookMgr. SetApplicablePriceBooks (priceBooks: PriceBook...) set pricebooks into the user session.
It works when the site's currency is USD, but when the currency is not USD, e.g. MYR, I still couldn't get ProductPriceModel.
Then I tried to set the currency in session to MYR, but failed. The error message is as follows:
**Wrapped java.lang.IllegalArgumentException: Illegal attempt to set session currency to currency 'MYR' which is not allowed by current site:Sites-Site Allowed currencies for the site are: [USD]**
Do you have any other Suggestions?

Thanks & Regards,
Tony

**Comment by zlatinz** [comment]
I would suggest as a workaround to try using OCAPI. I know it is not the greatest developer experience, but I do not have other ideas at the moment.

List of allowed currencies per site
https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.3/shop/Resources/Site.html#id1439497437__id-992161710

https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.3/shop/Resources/Products.html#id-918841120__id1435864274
You should be able to specify currency you are interested in as a currency URL parameter

**Comment by zlatinz** [comment]
Please raise a support ticket for this to be investigated by our support team.

**Comment by TonyRlq** [comment]
> Please raise a support ticket for this to be investigated by our support team.

Ok, I will raise a support ticket, thanks a lot for your reply.

**Comment by b-pupavac** [comment]
Hi @TonyRlq any update on this? I hit the same issue as you did.

**Comment by pkovalyov** [comment]
Hello guys,

We have a same issue on our site, support case is opened, and on investigation from SFCC side now.
Case number 00349354
Please mention this in your support case, may this will help to get answer faster.

Thanks.
